tflog/provider_test.go

```diff
@@ -1199,6 +1488,57 @@ func TestMaskMessageRegexes(t *testing.T) {
 	}
 }
 
+// Reference: https://github.com/hashicorp/terraform-plugin-log/issues/131
+func TestMaskMessageRegexes_NewContext(t *testing.T) {
+	t.Parallel()
+
+	var outputBuffer bytes.Buffer
+
+	originalCtx := context.Background()
+	originalCtx = loggertest.ProviderRoot(originalCtx, &outputBuffer)
+	originalCtx = tflog.MaskMessageRegexes(originalCtx, regexp.MustCompile("original"))
+
+	newCtx := tflog.MaskMessageRegexes(originalCtx, regexp.MustCompile("new"))
+
+	tflog.Trace(originalCtx, "original should be masked")
+	tflog.Trace(originalCtx, "new should be preserved")
+	tflog.Trace(newCtx, "new should be masked")
```

Comment:
Is it worth using different log levels for `originalCtx` and `newCtx` in order to be able to differentiate between them in the `expectedOutput`?

Reply:
Sure thing -- will adjust